tests/integration/main-test.js

```diff
@@ -171,5 +200,42 @@ module('Integration | main', function(hooks) {
 
     dogs = changeset.get('dogs');
     assert.equal(dogs.length, 2, 'changeset has 2 dogs');
+  }
+
+  test('it can rollback hasMany', async function(assert) {
+    await testRollbackHasMany.call(this, assert, 'user');
+  });
+
+  test('it can rollback sync hasMany', async function(assert) {
+    await testRollbackHasMany.call(this, assert, 'sync-user');
+  });
+
+  async function testInitiallyEmptyRelationships(assert, userType) {
+    let profile = this.store.createRecord('profile');
+    let user =  this.store.createRecord(userType);
+
+    let changeset = new Changeset(user);
+
+    changeset.set('profile', profile);
+    const dogs = [
+      this.store.createRecord('dog'),
+      this.store.createRecord('dog', { breed: 'Münsterländer' })
+    ];
+
+    changeset.set('dogs', dogs);
+
+    changeset.execute();
+
+    assert.equal(user.get('profile.firstName'), 'Bob', 'Profile is set on user');
+    assert.equal(user.get('dogs.firstObject.breed'), 'rough collie');
+    assert.equal(user.get('dogs.lastObject.breed'), 'Münsterländer');
```

Comment:
FYI I don't think these array accessors work (at the moment at least)

Reply:
Well, they work for EmberArray and `dogs` is an EmberArray.

Reply:
Oh you are right...we are passing in Ember's getter...